### DISCUSSION TITLE
How to integrate Teleport as an SP to Airlock IAM

### DISCUSSION THREAD

**Discussion by Aureliolo** [gravitational/teleport#33358] [Show and tell]
There are a few challenges when integrating it into Airlock IAM as Teleport requires multiple changes to the default settings in IAM. 

Start with creating a normal oauth 2 / oidc Server inside of Airlock IAM. Inside of the created oauth 2 server create the usual plugins and meta data links which are by default needed as well as the authorization flows. Since this is highly dependent on your environment im not gonna go into detail of that here. 

The challenges specifically and tings you have to configure ontop are the following: 

- Make sure you have a signing server active for that oauth flow
- The JWT token has to be signed, by default tokens are unsigned which makes Teleport not accept it. Create a Keystore for Teleport in the IAM and then a JWT access token signature inside IAM. 
-  Same with the oidc ID token, it also needs a signature so it has to be signed. (make sure to select the same encryption method as on the signing server as in our instance the default was a different one)
- you can do a normal roles to claim transformation on the auth flow so that teleport can grab those and assign teleport toles.

**Comment by webvictim** [comment]
Thanks for sharing!
